src/controllers/controller.bar.js

```diff
@@ -397,17 +407,23 @@ module.exports = function(Chart) {
 						imeta.stack === stack &&
 						imeta.controller.getValueScaleId() === scale.id &&
 						chart.isDatasetVisible(i)) {
+						yStackValue = scale.parseValue(datasets[i].data[index]);
 
-						ivalue = scale.getRightValue(datasets[i].data[index]);
-						if ((value < 0 && ivalue < 0) || (value >= 0 && ivalue > 0)) {
+						if (yStackValue.min === yStackValue.max) {
+							ivalue = yStackValue.val;
+						} else {
+							ivalue = yStackValue.val <= 0 ? yStackValue.min : yStackValue.max;
```

Comment:
`yStackValue.val <= 0` will never be true so this can just be `ivalue = yStackValue.max`

Reply:
pls, check the previous comment with "mirror effect" for negative and positive values. Its the same here. yStackValue.val <= 0 in case of negative number, so it should be as it is. I have changed a bit this part of code, so it look close to code where we calculate start point.